Sensus.Shared.iOS/Callbacks/UILocalNotifications/UILocalNotificationNotifier.cs

```diff
@@ -70,8 +70,20 @@ public void IssueNotificationAsync(string title, string message, string id, bool
                     UserInfo = notificationInfo
                 };
 
+                // if a protocol ID has been passed to the method, check the protocol's Notification Alert Exclusion Windows
+                // to determine whether to vibrate and play sound. if protocolId is null, just use the vibrateAndPlaySound parameter.
+                if (protocolId != null)
+                {
+                    var runningProtocol = SensusServiceHelper.Get().GetRunningProtocolById(protocolId);
+
+                    if (runningProtocol != null)
+                        foreach (Window window in runningProtocol.NotificationAlertExclusionWindowsList)
+                            if (window.EncompassesCurrentTime())
+                                vibrateAndPlaySound = false;
```

Comment:
The above code should be checking the `FireDate` time, not the current time. This means that the method you create in `Notifier` per previous comment will need to take in a DateTime parameter indicating the time that should be checked for exclusion.